core/src/main/scala/io/gearpump/transport/netty/Client.scala

```diff
@@ -200,11 +200,11 @@ object Client {
     }
   }
 
-  class ClientPipelineFactory(name: String, decoder: MessageDecoder, encoder: MessageEncoder) extends ChannelPipelineFactory {
+  class ClientPipelineFactory(name: String, conf: NettyConfig) extends ChannelPipelineFactory {
```

Comment:
He wants factory produce encoder decoder internally on demand. There is a thread unsafe issue.